src/main/java/uk/gov/pay/ledger/transaction/state/TransactionState.java

```diff
@@ -46,10 +54,34 @@ public String getMessage() {
         return message;
     }
 
-    private static final Map<SalientEventType, TransactionState> EVENT_TYPE_TRANSACTION_STATE_MAP = Map.of(
-            SalientEventType.PAYMENT_CREATED, CREATED,
-            SalientEventType.AUTHORISATION_SUCCESSFUL, SUBMITTED
-    );
+    private static final Map<SalientEventType, TransactionState> EVENT_TYPE_TRANSACTION_STATE_MAP =
+            ImmutableMap.<SalientEventType, TransactionState>builder()
```

Comment:
`Map.of` returns an immutable map (`ImmutableMap` was a google util for pre Java 11 times)

Reply:
`Map.of` has a size limit for the number of pairs. As you've suggested I'll use `Map.ofEntries` in favour of an external library